invoices/invoiceregistry.go

```diff
@@ -127,13 +143,26 @@ type invoiceEvent struct {
 	invoice *channeldb.Invoice
 }
 
-// invoiceEventNotifier is the dedicated goroutine responsible for accepting
+// invoiceEventLoop is the dedicated goroutine responsible for accepting
 // new notification subscriptions, cancelling old subscriptions, and
 // dispatching new invoice events.
-func (i *InvoiceRegistry) invoiceEventNotifier() {
+func (i *InvoiceRegistry) invoiceEventLoop() {
 	defer i.wg.Done()
 
+	// Set up a heap for htlc auto-releases.
+	autoReleaseHeap := &releaseHeap{}
+	heap.Init(autoReleaseHeap)
+
 	for {
+		// If there is something to release, set up a release tick
+		// channel.
+		var nextReleaseTick <-chan time.Time
+		if autoReleaseHeap.Len() > 0 {
```

Comment:
this would make a great addition to the `ticker` package 🤓 `ticker.Heap`?

Reply:
Yes it would, but I prefer to do this when the second release heap is introduced to see how it is best abstracted.

Reply:
Cherry picked prio queue from #3694